pinot-core/src/main/java/org/apache/pinot/core/common/evaluators/DefaultJsonPathEvaluator.java

```diff
@@ -90,7 +90,7 @@ public <T extends ForwardIndexReaderContext> void evaluateBlock(int[] docIds, in
         }
       }
     } else {
-      switch (reader.getValueType()) {
+      switch (reader.getStoredType()) {
```

Comment:
Secondly, if we make this change do we still need `getValueType` anywhere in the code to get the actual logical datatype of the column ? I guess no because ForwardIndexReader and it's users should only need to work with storedType except for DefaultIndexReaderProvider code ?

Also, for columns like BOOLEAN  / TIMESTAMP, this change should have ideally failed some tests whose storedType is INT / LONG because now the return value is different

Reply:
Checked the code in `DefaultIndexReaderProvider`. It correctly uses `storedType` to instantiate forward index reader and then in the constructor of the actual reader, it passes `storedType` even though the constructor says `valueType`. So, this is another reason why this cleanup / rename is good. This is also why no tests failed because of this change.

Reply:
Forward index only knows the stored type of the value, and it doesn't know the logical type of the value.
For the `JSON` branch under the switch, I believe they are never hit, and `getStoredType()` should never return `JSON`. We can try removing the `JSON` branch and see if all the tests still pass

Reply:
@Jackie-Jiang I commented out the `JSON` branch under the switch and ran all the tests under `pinot-core`. No test failures. Checked that `JsonExtractScalarTest` runs fine. Is there any additional testing I can do to validate that this `JSON` path is indeed never run?

Should I remove the `JSON` branch under these switch statements as part of this change?
cc @siddharthteotia

Reply:
I think that is good enough but let's update the PR with that change and let's get a clean build to assess.